### PR TITLE
fixes for #3729 and making sure we dont reply on stderror and just vs…

### PR DESCRIPTION
**Pull request by acesiddhu**
fixes for #3729 and making sure we dont reply on stderror and just vstest.console.exe error code for failing the tasks

### CODE REVIEW

**Comment by kaadhina** [comment]
Please specify scenarios tested. Changes LGTM

**Comment by acesiddhu** [comment]
1) test where test will print something in error stream. with and without the change
2) above scenario and a failing test with ignore test failure flag
3) no runsettings scenario
4) normal passing scenario

File: Tasks/VsTest/vstest.ts [hunk 1]
```diff
@@ -136,15 +136,21 @@ function getVstestArguments(settingsFile: string, tiaEnabled: boolean): string[]
...
     }
-    if (settingsFile && pathExistsAsFile(settingsFile)) {
-        argsArray.push("/Settings:" + settingsFile);
-        utils.Helper.readFileContents(settingsFile, "utf-8").then(function (settings) {
-        tl.debug("Running VsTest with settings : " + settings);
-        });
+    if (settingsFile) {
+        if (pathExistsAsFile(settingsFile)) {
+            argsArray.push("/Settings:" + settingsFile);
+            utils.Helper.readFileContents(settingsFile, "utf-8").then(function (settings) {
```
Comment (cltshivash) [inline Tasks/VsTest/vstest.ts]: can't we use await pattern here.. ?

File: Tasks/VsTest/vstest.ts [hunk 2]
```diff
@@ -434,10 +440,15 @@ function executeVstest(testResultsDirectory: string, parallelRunSettingsFile: st
 
     tl.cd(workingDirectory);
     var ignoreTestFailures = vstestConfig.ignoreVstestFailure && vstestConfig.ignoreVstestFailure.toLowerCase() === "true";
-    vstest.exec(<tr.IExecOptions>{ failOnStdErr: !ignoreTestFailures })
+    vstest.exec(<tr.IExecOptions>{ ignoreReturnCode: ignoreTestFailures, failOnStdErr: false })
         .then(function (code) {
```
Comment (cltshivash) [inline Tasks/VsTest/vstest.ts]: is there some place we log the exit code..? if not please do add..

File: Tasks/VsTest/vstest.ts [hunk 3]
```diff
@@ -434,10 +440,15 @@ function executeVstest(testResultsDirectory: string, parallelRunSettingsFile: st
...
             cleanUp(parallelRunSettingsFile);
-            defer.resolve(code);
+            if (ignoreTestFailures === true) {
+                defer.resolve(0); // ignore failures.
+            }
+            else {
+                defer.resolve(code);
+            }
         })
         .fail(function (err) {
```
Comment (cltshivash) [inline Tasks/VsTest/vstest.ts]: can we please log it here...?

File: Tasks/VsTest/vstest.ts (hunk 2 above)
Comment (cltshivash) [inline Tasks/VsTest/vstest.ts]: should we move to a async/await and catch the error model ..?

**Pull request merged**